src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -234,6 +236,41 @@ private static void addAwsConfigGroup(final ConfigDef configDef) {
             AWS_S3_REGION_CONFIG
         );
 
+        configDef.define(
+                AWS_S3_PART_SIZE,
+                Type.INT,
+                S3OutputStream.DEFAULT_PART_SIZE,
+                new ConfigDef.Validator() {
+                    @Override
+                    public void ensureValid(final String name, final Object value) {
+                        if (value == null) {
+                            throw new ConfigException(name, null, "Part size must be non-null");
+                        }
+                        final var number = (Number) value;
+                        if (number.longValue() < S3OutputStream.DEFAULT_PART_SIZE) {
+                            throw new ConfigException(
+                                    name,
+                                    value,
+                                    "Part size must be at least: " + S3OutputStream.DEFAULT_PART_SIZE + " bytes (5MB)"
+                            );
+                        }
+                        if (number.longValue() > Integer.MAX_VALUE) {
+                            throw new ConfigException(
+                                    name,
+                                    value,
+                                    "Part size must be no more: " + Integer.MAX_VALUE + " bytes (~2GB)"
+                            );
+                        }
+                    }
+                },
```

Comment:
Basically stealing the comment from another PR: Why not
```suggestion
        ConfigDef.Range.between(S3OutputStream.DEFAULT_PART_SIZE, Integer.MAX_VALUE),
```

Reply:
true true. i forget about `Range` will change